Snapshot config to define column names (#6146)
dbt-labs/docs.getdbt.com: website/docs/reference/snapshot-configs.md

```

</File>

<VersionBlock firstVersion="1.9">

> #### Define column names in snapshot table

Reviewer: ```suggestion
#### Customize meta column names in snapshot table
```

> Use the following configuration to define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).

Reviewer: ```suggestion
Use the following configuration to customize the names of the meta columns dbt creates in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).
```

> You can use the following config to define the names of the columns that are created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).

Reviewer 2: ```suggestion
Use the following configuration to define the names of the columns created in the [snapshot table](/docs/build/snapshots#snapshot-meta-fields).
```

> <File name='dbt_project.yml'>

Reviewer: let's also show the example from the issue on how to do this in the config block:

```
{{
    config(
      target_database='analytics',
      target_schema='snapshots',
      unique_key='id',

      strategy='timestamp',
      updated_at='updated_at',
      
      snapshot_meta_column_names={
        dbt_valid_from: start_date,
        dbt_valid_to: end_date,
        dbt_scd_id: scd_id,
        dbt_updated_at: updated_date
      }
    )
}}
```

Reviewer: we have some tab thing for other examples so you can see how to write a config in different places

```yml

> snapshots:

Reviewer: Let's use the full example from the issue!

```
snapshots:
  my_project:
    +snapshot_meta_column_names={
      dbt_valid_from: start_date,
      dbt_valid_to: end_date,
      dbt_scd_id: scd_id,
      dbt_updated_at: updated_date
    }
```

  my_project:
    +snapshot_meta_column_names ={
      dbt_valid_from: example_valid_from,
      dbt_valid_to: example_valid_to
    }

```

</File>

- When you run your snapshots, dbt handles the renames from `dbt_valid_from` to `example_valid_from` for you.

> - Update your config to:

Reviewer 2: ```suggestion
- If you make subsequent updates to the config to rename the column, dbt manages the migration of column names for each configuration change. For example:
```

> - If you make subsequent updates to the config to rename the column, dbt manages the migration of column names for each configuration change. For example:

Reviewer: This is not true.

See issue -> https://github.com/dbt-labs/docs.getdbt.com/issues/6112#issuecomment-2374659927

> we aren't doing anything for migration, so if a user renames the columns for a snapshot without updating an existing table, they will get an error. The recommendation is to use these settings for new snapshots, or to arrange to update their existing tables prior to committing a column name change.

<File name='dbt_project.yml'>

```yml

snapshots:
  my_project:
    +snapshot_meta_column_names ={
      dbt_valid_from: test_valid_from,
      dbt_valid_to: test_valid_to
    }

```

</File>

> - When you run your snapshots, dbt handles the renames from `example_valid_from` to `test_valid_from` for you.

Reviewer: this is not true

> - When you run your snapshots, dbt handles the renames from `example_valid_from` to `test_valid_from` for you.

Reviewer: same comment as above - this is not true

> You can read through the section on [Defining column names](/docs/build/snapshots#define-column-names-in-snapshot-table) for more information.

Reviewer 2: the link doesn't take me to the right place? or for some reason i didn't see the 'define column names' section?
